src/System.Linq/tests/ZipTests.cs

```diff
@@ -11,6 +11,224 @@ namespace System.Linq.Tests
+            IEnumerable<(int, int)> expected = new (int, int)[] { };
+
+            Assert.Equal(expected, first.Zip(second));
+        }
+
+        [Fact]
+        public void Zip2_SecondEmptyFirstMany()
+        {
+            IEnumerable<int> first = new int[] { 1, 2, 3 };
+            IEnumerable<int> second = new int[] { };
+            IEnumerable<(int, int)> expected = new (int, int)[] { };
+
+            Assert.Equal(expected, first.Zip(second));
+        }
+
+        [Fact]
+        public void Zip2_FirstAndSecondSingle()
+        {
+            IEnumerable<int> first = new int[] { 1 };
+            IEnumerable<int> second = new int[] { 2 };
+            IEnumerable<(int, int)> expected = new (int, int)[] { (1, 2) };
+
+            Assert.Equal(expected, first.Zip(second));
+        }
+
+        [Fact]
+        public void Zip2_FirstAndSecondEqualSize()
+        {
+            IEnumerable<int> first = new int[] { 1, 2, 3 };
+            IEnumerable<int> second = new int[] { 2, 3, 4 };
+            IEnumerable<(int, int)> expected = new (int, int)[] { (1, 2), (2, 3), (3, 4) };
+
+            Assert.Equal(expected, first.Zip(second));
+        }
+
+        [Fact]
+        public void Zip2_SecondOneMoreThanFirst()
+        {
+            IEnumerable<int> first = new int[] { 1, 2 };
+            IEnumerable<int> second = new int[] { 2, 4, 8 };
+            IEnumerable<(int, int)> expected = new (int, int)[] { (1, 2), (2, 4) };
+
+            Assert.Equal(expected, first.Zip(second));
+        }
+
+
+        [Fact]
+        public void Zip2_SecondManyMoreThanFirst()
+        {
+            IEnumerable<int> first = new int[] { 1, 2 };
+            IEnumerable<int> second = new int[] { 2, 4, 8, 16 };
+            IEnumerable<(int, int)> expected = new (int, int)[] { (1, 2), (2, 4) };
+
+            Assert.Equal(expected, first.Zip(second));
+        }
+
+        [Fact]
+        public void Zip2_FirstOneMoreThanSecond()
+        {
+            IEnumerable<int> first = new int[] { 1, 2, 3 };
+            IEnumerable<int> second = new int[] { 2, 4 };
+            IEnumerable<(int, int)> expected = new (int, int)[] { (1, 2), (2, 4) };
+
+            Assert.Equal(expected, first.Zip(second));
+        }
+
+
+        [Fact]
+        public void Zip2_FirstManyMoreThanSecond()
+        {
+            IEnumerable<int> first = new int[] { 1, 2, 3, 4 };
+            IEnumerable<int> second = new int[] { 2, 4 };
+            IEnumerable<(int, int)> expected = new (int, int)[] { (1, 2), (2, 4) };
+
+            Assert.Equal(expected, first.Zip(second));
+        }
+
+        [Fact]
+        public void Zip2_ForcedToEnumeratorDoesntEnumerate()
+        {
+            var iterator = NumberRangeGuaranteedNotCollectionType(0, 3).Zip(Enumerable.Range(0, 3));
+            // Don't insist on this behaviour, but check it's correct if it happens
+            var en = iterator as IEnumerator<int>;
+            Assert.False(en != null && en.MoveNext());
```

Comment:
I'm not understanding this test.  Why would the `IEnumerable<(int, int)>` ever be an `IEnumerator<int>`?

Reply:
I honestly have no idea about this one. There's a corresponding test for the old Zip method. I just deleted this one because I don't know what it's trying to test.